nucypher/cli/actions.py

```diff
@@ -304,31 +303,66 @@ def make_cli_character(character_config,
 
 
 def select_stake(stakeholder, emitter) -> Stake:
-    enumerated_stakes = dict(enumerate(stakeholder.stakes))
-    painting.paint_stakes(stakes=stakeholder.stakes, emitter=emitter)
+    stakes = stakeholder.all_stakes
+    enumerated_stakes = dict(enumerate(stakes))
+    painting.paint_stakes(stakes=stakes, emitter=emitter)
     choice = click.prompt("Select Stake", type=click.IntRange(min=0, max=len(enumerated_stakes)-1))
     chosen_stake = enumerated_stakes[choice]
     return chosen_stake
 
 
-def select_client_account(emitter, blockchain, prompt: str = None, default=0) -> str:
+def select_client_account(emitter,
+                          provider_uri: str,
+                          prompt: str = None,
+                          default: int = 0,
+                          registry=None,
+                          show_balances: bool = True
+                          ) -> str:
+    """
+    Note: Setting show_balances to True, causes an eager contract and blockchain connection.
```

Comment:
Even if `show_balances=False`, there's an eager connection to the blockchain to retrieve the list of accounts, no?